flang/test/Driver/Inputs/config-l.cfg

```diff
@@ -0,0 +1 @@
+-ffast-math -Wl,--as-needed | -lm -Wl,-Bstatic -lhappy -Wl,-Bdynamic
```

Comment:
Can we check that this works for multiple line config files? E.g. I would expect:
```
-ffast-math | -lm
-Wl,--as-needed | -Wl,-Bstatic -lhappy
| -Wl,-Bdynamic
```
to end up being essentially the same as what is here on one line. I.e. `-ffast-math` and `-Wl,--as-needed` should appear before user-specified options, and `-lm` `-Wl,-Bstatic` `-lhappy` and `-Wl,-Bdynamic` to appear after.

I gave this a quick test and it seems like doing this on multiple lines like above changes the order; I think everything after the first pipe _in the file_ is considered to be added to the end, whereas I'd expect it to be on a line-by-line basis. I guess this might need a change in `readConfigFile()`.

Reply:
Since for ` llvm::cl::ExpansionContext::readConfigFile()` a newline is just a separator, this can't be achieved by doing an enhancement in a code very distant from here. Seems now the 'pipe' idea can be buried and we're back to square one. I need to invent something different.

Reply:
We could prefix those linker flags with some character, e.g. '@', it would work like that:

```
-ffast-math @-lm
-Wl,--as-needed @-Wl,-Bstatic -lhappy
@-Wl,-Bdynamic
```

Reply:
Could we just pick a character to prefix each argument with that indicates that that argument should come at the end? `$` maybe?
e.g. 
```
-ffast-math $-lm 
-Wl,--as-needed, $-Wl,-Bstatic 
$-lhappy $-Wl,-Bdynamic
```
It's a bit messy compared to your pipe idea but maybe easier to implement?

Reply:
mid-air collision. yeah, either '$' or '@' would do :)

Reply:
I'm indifferent to what symbol we choose, but I believe `@` is already used for including other config files, so we should avoid that to avoid having a collision there.

Reply:
Seems neither '$' nor '@' is the best choice, '$' looks like shell variable expansion, so I'll do `^`, e.g., '^-Wl,-Bstatic`.

Reply:
Done.